src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1410,41 +1410,80 @@ private function dumpValue($value, $interpolate = true)
             }
 
             return sprintf('array(%s)', implode(', ', $code));
-        } elseif ($value instanceof ServiceClosureArgument) {
-            $value = $value->getValues()[0];
-            $code = $this->dumpValue($value, $interpolate);
+        } elseif ($scope = $value instanceof ArgumentInterface) {
+            $scope = array($this->definitionVariables, $this->referenceVariables, $this->variableCount);
```

Comment:
Reusing the same variable to store a boolean or an array is harder to follow. Can't we put the `try/finally` inside this if instead, which will allow to only store an array in `$scope` ?